### PR TITLE
fix: action provider compilation

### PR DESCRIPTION
**Pull request by octonato**

### CODE REVIEW

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala [hunk 1]
```diff
@@ -294,7 +294,7 @@ object ActionServiceSourceGenerator {
         |}
         |
         |class ${service.providerName} private(actionFactory: ActionCreationContext => ${service.className},
-        |                                      options: ActionOptions)
+        |                                      val options: ActionOptions)
```
Comment (patriknw) [inline codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala:297]: add `override` to make it more clear

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala [hunk 2]
```diff
@@ -294,7 +294,7 @@ object ActionServiceSourceGenerator {
...
         |  extends ActionProvider[${service.className}] {
         |
```
Comment (patriknw) [inline codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala:299]: add `def withOptions` while you are at it

Comment (octonato) [reply codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala:299]: it's there already, line 310

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala (hunk 1 above)
Comment (octonato) [reply codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala:297]:
```suggestion
        |                                      override val options: ActionOptions)
```

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala (hunk 2 above)
Comment (patriknw) [reply codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala:299]: ah, thanks

File: codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/ActionServiceSourceGeneratorSuite.scala [hunk 3]
```diff
@@ -182,7 +182,7 @@ class ActionServiceSourceGeneratorSuite extends munit.FunSuite {
          |}
          |
          |class MyServiceActionProvider private(actionFactory: ActionCreationContext => MyServiceAction,
-         |                                      options: ActionOptions)
+         |                                      val options: ActionOptions)
```
Comment (patriknw) [inline codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/ActionServiceSourceGeneratorSuite.scala:185]: `override`

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**